core/server/data/migrations/versions/4.42/2022-03-21-17-17-add.js

```diff
@@ -0,0 +1,20 @@
+const {addTable} = require('../../utils');
+
+module.exports = addTable('newsletters', {
+    id: {type: 'string', maxlength: 24, nullable: false, primary: true},
+    name: {type: 'string', maxlength: 191, nullable: false},
+    description: {type: 'string', maxlength: 2000, nullable: true},
+    sender_name: {type: 'string', maxlength: 191, nullable: false},
+    sender_email: {type: 'string', maxlength: 191, nullable: false, validations: {isEmail: true}},
+    sender_reply_to: {type: 'string', maxlength: 191, nullable: false, validations: {isEmail: true}},
+    default: {type: 'bool', nullable: false, defaultTo: false},
+    status: {type: 'string', maxlength: 50, nullable: false, defaultTo: 'active'},
+    recipient_filter: {
+        type: 'text',
+        maxlength: 1000000000,
+        nullable: false,
+        defaultTo: 'status:-free'
```

Comment:
This field would be a filter to prevent non-matching users from being able to subscribe to the newsletter. I believe posts on a newsletter would inherit/copy this field indeed.

You're right we might want to leave the default open to all users, I'm gonna make it empty